April CQA work for MTA 8.1 Release Notes DITA fixes (#356)
migtools/mta-documentation: docs/topics/release-notes-topics/ref_fixed-issues-8-0.adoc

= Fixed issues

[role="_abstract"]

> Review the following fixed issues that have a significant impact in {ProductFirstRef} ({ProductShortName}) version 8.0.0. 

Reviewer: ```suggestion
Review issues that have been fixed in {ProductFirstRef} ({ProductShortName}) version 8.0.0. 
```

{ProductShortName} merges the default `values.yaml` file with the discover `values.yaml` file when generating a deployment manifest::
Before this update, when you generated a deployment manifest by using the Helm template, {ProductShortName} ignored the default values in the `values.yaml` file packaged with the Helm chart. As a result, {ProductShortName} used only the values from the `values.yaml` file created after running the `mta-cli discover` command. With this update, {ProductShortName} merges the default `values.yaml` file with the discover `values.yaml` file.
+
link:https://issues.redhat.com/browse/MTA-5793[MTA-5793]

{ProductShortName} CLI no longer fails to detect dependencies for Gradle projects::

> Before this update, when you ran an application analysis for Gradle projects in containerless mode, {ProductFullName} command-line interface (CLI) did not detect Gradle dependencies. As a result, the analysis report did not list the dependencies and displayed an error message in the analysis logs instead. With this update, {ProductShortName} CLI detects Gradle dependencies after the analysis is complete.

Reviewer: Same thing as above.
```suggestion
Before this update, when you ran an application analysis for Gradle projects in containerless mode, {ProductFullName} command-line interface (CLI) did not detect Gradle dependencies. Consequently, the analysis report did not list the dependencies and displayed an error message in the analysis logs instead. With this update, {ProductShortName} CLI detects Gradle dependencies after the analysis is complete.
```

+
link:https://issues.redhat.com/browse/MTA-4033[MTA-4033]

{ProductShortName} UI no longer fails to analyze projects built with Gradle version 7 and later::
Before this update, when you analyzed an application built with Gradle in `Source code + dependencies` mode in the {ProductFullName} user interface (UI), the analysis  failed if you had Gradle version 7 or later installed. With this update, you can analyze applications built with Gradle 7 or later in the {ProductShortName} user interface.
+
link:https://issues.redhat.com/browse/MTA-5907[MTA-5907]

The discovery manifest lists all defined web processes in the application::

> Before this update, the `mta-cli discover` command only moved an application-level process with the process type explicitly defined to the `processes` block in the discovery manifest. As a result, the discovery manifest did not list implicit application-level processes in the `processes` block. This resulted in the violation of the recommended clarity and proper configuration practices, especially if Cloud Foundry application manifest includes multiple process types. With this update, `mta-cli discover` moves both implicit and explicit application-level process types to the `processes` block in the discovery manifest.

Reviewer: Tense agreement + same things as above (not a result, a consequence)
```suggestion
Before this update, the `mta-cli discover` command only moved an application-level process with the process type explicitly defined to the `processes` block in the discovery manifest. Consequently, the discovery manifest did not list implicit application-level processes in the `processes` block. This resulted in the violation of the recommended clarity and proper configuration practices, especially if Cloud Foundry application manifest included multiple process types. With this update, `mta-cli discover` moves both implicit and explicit application-level process types to the `processes` block in the discovery manifest.
```

[role="_abstract"]

> {ProductFirstRef} ({ProductShortName}) version 8.0.0 provides the following fixed issues and other problems that have a significant impact. 

Reviewer 2: What 'other problems' does it provide? How are they different from fixed issues and why is it important to differentiate the two?

Author: You don't want to know!

The discovery manifest lists all defined web processes in the application::

> Before this update, the `mta-cli discover` command only moved an application-level process with the process type explicitly defined to the `processes` block in the discovery manifest. As a result, the discovery manifest did not list implicit application-level processes in the `processes` block. This resulted in the violation of the recommended clarity and proper configuration practices, especially if CloudFoundry application manifest includes multiple process types. With this update, `mta-cli discover` moves both implicit and explicit application-level process types to the `processes` block in the discovery manifest.

Reviewer 2: I noticed that in the docs/topics/release-notes-topics/ref_fixed-issues-8-0-1.adoc file line 24 Cloud Foundry is spelled differently (2 words). So I assume this is the correct spelling? Fix if it occurs elsewhere too.

```suggestion
Before this update, the `mta-cli discover` command only moved an application-level process with the process type explicitly defined to the `processes` block in the discovery manifest. As a result, the discovery manifest did not list implicit application-level processes in the `processes` block. This resulted in the violation of the recommended clarity and proper configuration practices, especially if Cloud Foundry application manifest includes multiple process types. With this update, `mta-cli discover` moves both implicit and explicit application-level process types to the `processes` block in the discovery manifest.
```
